docs/features/kubernetes.rst

```diff
@@ -0,0 +1,56 @@
+Kubernetes
+==============
+
+This feature was requested as part of `Issue 345 <https://github.com/ThreeMammals/Ocelot/issues/345>`_ . to add support for kubernetes's service discovery provider. 
+
+The first thing you need to do is install the NuGet package that provides kubernetes support in Ocelot.
+
+``Install-Package Ocelot.Provider.Kubernetes``
+
+Then add the following to your ConfigureServices method.
+
+.. code-block:: csharp
+
+    s.AddOcelot()
+     .AddKubernetes();
+
+If you have services deployed in kubernetes you will normally use the naming service to access them.
+
+The following example shows how to set up a ReRoute that will work in kubernetes. The most important thing is the ServiceName which is made up of the 
+kubernetes service name. We also need to set up the ServiceDiscoveryProvider in 
+GlobalConfiguration. The example here shows a typical configuration. It assumes kubernetes api server is running on 192.168.0.13 and that the api service is on port 443.
+
+The example below is taken from the samples folder so please check it if this doesnt make sense!
+
+.. code-block:: json
+
+    {
+  "ReRoutes": [
+    {
+      "DownstreamPathTemplate": "/api/values",
+      "DownstreamScheme": "http",
+      "UpstreamPathTemplate": "/values",
+      "ServiceName": "downstreamservice",
+      "UpstreamHttpMethod": [ "Get" ],
+      "QoSOptions": {
```

Comment:
i would try to keep the configuration as simple as possible, no qos, no cache, nothing that is not entirely necessary for this feature to work

Reply:
agree